torch/nn/modules/lazy.py

```diff
@@ -0,0 +1,148 @@
+import itertools
+
+import torch
+from ..parameter import UninitializedParameter
+
+
+class LazyModuleMixin:
+    r"""A mixin for modules that lazily initialize parameters.
+
+    Unlike regular modules, subclasses of this module can initialize
+    buffers outside of the constructor (``__init__``).
+    This allows you to, for example, initialize parameters in ``forward``
+    method to determine the shape of the weight based on the initial input.
+    Be sure to run "dummy" forward once to initialize all parameters in case
+    that you need to access the weights before the actual real ``forward`` pass.
+
+    Note that lazy modules cannot validate if the shape is correct during
+    deserialization.  Also note that the initial weights values may become different
+    from the original (non-lazy) module even if the random seed is manually
+    configured, as the order of initialization is different from the original
+    one; especially, ``module.cuda()`` may cause the initialization to run on
+    a GPU.
+
+    The default value of lazy parameters are ``UninitializedParameter()``.
+
+    .. note:: A `LazyModule`, or a module containing a `LazyModule` requires an explicit
+    dummy forward call in the following scenarios.
+
+
+    >>> lazy_module = LazyLinear(10)
```

Comment:
this code sample seems out of place here, it's not a scenario requiring explicit forward